packages/story-editor/src/app/pageCanvas/pageCanvasProvider.js

```diff
@@ -0,0 +1,205 @@
+/*
+ * Copyright 2022 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+/**
+ * External dependencies
+ */
+import PropTypes from 'prop-types';
+import { useCallback, useRef } from '@googleforcreators/react';
+
+/**
+ * Internal dependencies
+ */
+import storyPageToCanvas from '../../utils/storyPageToCanvas';
+import { getAccessibleTextColorsFromPixels } from '../../utils/contrastUtils';
+import useIdleQueue from '../../utils/useIdleTaskQueue';
+import { useStory } from '../story';
+import { STABLE_ARRAY } from '../../constants';
+import Context from './context';
+import getPixelDataFromCanvas from './getPixelDataFromCanvas';
+import usePageCanvasMap from './usePageCanvasMap';
+import usePageSnapshot from './usePageSnapshot';
+import PageCanvasCacheValidator from './pageCanvasCacheValidator';
+import getPageWithoutSelection from './getPageWithoutSelection';
+
+/**
+ * @typedef {import('../../../types').Page} Page
+ */
+
+function PageCanvasProvider({ children }) {
+  const queueIdleTask = useIdleQueue();
+  const [pageCanvasMap, actions] = usePageCanvasMap();
+  const { getSnapshotCanvas, setSnapshot } = usePageSnapshot();
```

Comment:
💭 was really helpful to chat with you about why these two caches are here. Would be awesome to get some documentation here (if there isn't any already).

I found it hard to understand why these were here until you walked me through it, so maybe others will too!